### DISCUSSION TITLE
Change BT Name on the fly?

### DISCUSSION THREAD

**Discussion by harryberlin** [pschatzmann/ESP32-A2DP#568] [Q&A] [answered]
Is it possible to change the Bluetooth Name on the fly, instead of using .end() and .start() again with the new name?

**Comment by pschatzmann** [comment, marked as answer]
You can try to implement this yourself: here is the [link to the GAP API](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/bluetooth/esp_gap_bt.html)

**Comment by harryberlin** [comment]
Did add simple methode to 'BluetoothA2DPSink'
```
void BluetoothA2DPSink::bt_rename(const char *name) {
  this->bt_name = name;
  esp_bt_dev_set_device_name(bt_name);
}
```

Update of name does work only work, if there is no connection at the moment.

**Comment by pschatzmann** [comment]
I would expect that if you are connected you can disconnect and set the discoverability to true
